Code-4-Community/speak-for-the-trees-frontend
# Add Drop Down Filter and Table View to Review Images Page

## Checklist


## Why

Resolves #317
Resolves #293

This ticket flushes out most of the work for image approval. Admin are able to filter, view and select unapproved site images. Functionality to approve and reject images will be implemented in the future. 

## This PR

The following changes are made:
- Adds a table component to view images `src/components/unapprovedImagesTable/index.tsx`
- Adds a filter component for unapproved images `src/components/unapprovedFilterImageControls/index.tsx`
- Adds filterSiteImages as a client request `src/api/protectedApiClient.ts`
- Adds tests for filterSiteImages `src/api/test/protectedApiClient.test.ts`
- Places the filter and table for unapproved images into `src/containers/reviewImages/index.tsx`

## Screenshots
<img width="1057" alt="Screenshot 2024-04-14 at 2 44 46 PM" src="https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/30560773/bf1914bc-b16f-4b25-9107-29bb16b7379e">

<img width="1164" alt="Screenshot 2024-04-14 at 2 43 29 PM" src="https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/30560773/a1ffd486-1969-49a4-abb5-90b6e5f37568">


## Verification Steps

The client calls for this change were tested in `src/api/test/protectedApiClient.test.ts.` All visual components were manually tested as follows.

1. Login as an admin
2. Ensure that the site_images table is populated in the speak-for-the-trees database with sample unapproved images
3. Navigate to your logged in icon -> Admin -> Review Images
4. Enter a date, site id, or neighborhood
5. Monitor as different images are made visible in the table
